### DISCUSSION TITLE
Update static page

### DISCUSSION THREAD

**Discussion by MohammadPCh** [vercel/next.js#16224] [Help]
Hi,
I'm developing a site about movies, an example URL is:
**https://example.com/movie/[id]/[name]**
for exmaple:
**https://example.com/movie/1/blabla**
I build each page in run time with set **fallback: true**, and the site has about **200K** movie page. Some times the content of one page needs to update. for example, I build 200,000 static pages, and one movie with id 60 was changed! I don't want to build all pages again, I just want to update a page.
I found that I can remove files of that page from build folder: (**blabla.json & blabla.html)** but it seems NextJS cached it in its memory and it serves the old files without building new files.
I know NextJs introduce **revalidate** for updating, but it is useless for this scenario because I don't want update page continuously. **I just want to update a page sometimes.**

In my opinion, it's nice to have an **API for purge a static page**, so the user doesn't need to build 200,000 pages to update one page.

**Comment by MohammadPCh** [comment]
Does anybody have an idea?

**Reply by benbachhuber** [reply]
@MohammadPCh Did you find a solution? This is something I was wondering about as well

**Reply by ysvahn** [reply]
Did you try to return your `initialProps` from `getStaticProps` with the `revalidate` flag ?

Like so : 

```javascript
    initialProps = await api.get();
    return { props: { ...initialProps }, revalidate: 1 };
```

Link to doc : https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

**Reply by benbachhuber** [reply]
But that would mean it would regenerate every second if we have a site with very high traffic right?

I guess there needs to be some functionality to actively trigger a regeneration of a specific static page.
Let's say we have a product page where the data changes every few days but whenever it changes the page should be updated soon.
- revalidate of 1 would mean lot's of regenerations which are not necessary
- a high revalidate value would mean a potentially outdated page

**Reply by MohammadPCh** [reply]
I post a feature request here: [https://github.com/vercel/next.js/issues/16488](https://github.com/vercel/next.js/issues/16488).
I suggest a solution, please watch it and let's talk there.
@BenBach @Kroonax
